pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -361,6 +384,52 @@ private boolean forceLog(BrokerResponse brokerResponse, long totalTimeMs) {
     return false;
   }
 
+
+  private void scoreResponse(BrokerResponse response, RequestStatistics stats) {
+
+    try {
+
+      if (response == BrokerResponseNative.EMPTY_RESULT || response == BrokerResponseNative.NO_TABLE_RESULT
+          || response.getNumServersResponded() == 0 || !response.getProcessingExceptions().isEmpty()) {
+        // error response
+        _brokerMetrics.addMeteredTableValue(stats.getTableName(), BrokerMeter.ERROR_RESPONSE, 1);
+        return;
+      }
+
+      if ((response.getNumServersQueried() > response.getNumServersResponded())
+          || stats.getFanoutType() == RequestStatistics.FanoutType.HYBRID
+          && response.getNumConsumingSegmentsQueried() <= 0 || response.isNumGroupsLimitReached()) {
+        // partial response
+        _brokerMetrics.addMeteredTableValue(stats.getTableName(), BrokerMeter.PARTIAL_RESPONSE, 1);
+        return;
+      }
+
+      TableConfig tableConfig = _tableConfigCache.get(stats.getTableName());
```

Comment:
note: stats.getTableName() gets raw table name. Which table config are you expecting to process below? I suppose realtime? I would imagine that freshness (in general) will also apply to offline tables. We are not impleemnting that now, I suppose?